modules/internal/ChapelBase.chpl

```diff
@@ -636,17 +636,73 @@ module ChapelBase {
   //
   // left and right shift on primitive types
   //
-  inline proc <<(a: int(?w), b: integral) return __primitive("<<", a, b);
-  inline proc <<(a: uint(?w), b: integral) return __primitive("<<", a, b);
 
-  inline proc >>(a: int(?w), b: integral) return __primitive(">>", a, b);
-  inline proc >>(a: uint(?w), b: integral) return __primitive(">>", a, b);
+  inline proc bitshiftChecks(a, b: integral) {
+    use HaltWrappers;
+
+    if b < 0 {
+      var msg = "Cannot bitshift " + a:string + " by " + b:string +
+                " because " + b:string + " is less than 0";
+      HaltWrappers.boundsCheckHalt(msg);
+    } else if b >= numBits(a.type) {
```

Comment:
I don't think `>=` is the right comparison, I think it should be `>` (By looking at the PR message it also looks like that's what you intended)

Reply:
No, sorry, the original intent is to use `>=`. Just for reference, this check was motivated by most C compilers classifying such as undefined behavior: 

https://wiki.sei.cmu.edu/confluence/display/c/INT34-C.+Do+not+shift+an+expression+by+a+negative+number+of+bits+or+by+greater+than+or+equal+to+the+number+of+bits+that+exist+in+the+operand

My rationale here is that IMO in Chapel, one should only encounter undefined behavior when opting into it explicitly (IE, via `--fast`).

The Chapel standard explicitly prohibits shifting by a negative value, but it does not mention this second check (which I would consider an oversight (?)).